Convert Version into a type-safe enum

texdc opened the pull request:
```php
use TinCan\Version;

$version = Version::v101();

assert($version->isLatest());
assert($version->isSupported());
assert($version->hasValue(Version::V101));
```
I've been digging through the other TinCan repos to better understand how the API is intended to be implemented.  Since PHP does not support the `enum` type without [an extension](http://php.net/manual/en/book.spl-types.php), I've used the `type-safe enum` pattern§ to better align `Version` with the other stacks.  It's not a `1:1` match, but it does improve stability by replacing version strings with immutable class constants and extends functionality by encapsulating version logic in a class object.  The trade-off is added complexity.  However, I've maintained BC with the existing methods.  While the new constants can be used to replace existing version strings, I would recommend leveraging the new functionality by adding a type-hint to the `asVersion()` method(s).
```php
public function asVersion(Version $aVersion);
```

> Note:  The `hhvm` build failed from what looks to be an invalid file path for the new cert.

> § Item 21: Replace enum constructs with classes, *Effective Java Programming Language Guide*, Joshua Bloch

brianjmiller commented on .travis.yml:
```diff
@@ -1,8 +1,16 @@
+matrix:
+  fast_finish: true
+  allow_failures:
+    - php: hhvm
```
I'd rather not just shut it off, I'm working on getting the Travis config updated and hoping that in doing so we'll be able to successfully build on hhvm again. This is likely to conflict with this change, in the future it'd be good to isolate changes to this kind of file from actual source code changes to avoid conflicts.

brianjmiller replied:
This can still merge cleanly but I have hhvm working so would rather not include it in the allow failures. Can you merge in master and to see if this still passes on Travis?

brianjmiller commented on src/Version.php:
```diff
@@ -17,19 +17,143 @@
+     * 
+     * @example $version = Version::V101();
+     * @param   string $aValue    the called method as a version value
+     * @param   array  $arguments unused argments passed to the method
```
Typo 'argments' -> 'arguments'.

brianjmiller commented:
@texdc I know this has been sitting out here like the other, please let me know if you think you can get back to it, otherwise I'll try to work it in, in the near future (this time).

texdc commented:
Sure.  I'll try to get to it no later than this weekend.  Is that acceptable?



> On Nov 17, 2015, at 3:29 PM, Brian J. Miller <notifications@github.com> wrote:
> 
> @texdc I know this has been sitting out here like the other, please let me know if you think you can get back to it, otherwise I'll try to work it in, in the near future (this time).
> 
> —
> Reply to this email directly or view it on GitHub.
>

brianjmiller commented:
@texdc given what we're paying you I'd say so ;-).

texdc commented:
@brianjmiller I've reverted the Travis config changes (should be on their own PR) and fixed the comment typo.  The failure looks unrelated.

You really want to fix my pay-scale!  :smile:

brianjmiller commented:
Because of #32 landing this now has conflicts, but it was already marked as failed tests, I'm working on rebasing the commit and running the Travis tests on the updated branch so I'm going to ultimately close this PR as unmerged, but the changes will land on master today (assuming the tests pass, or I can make them pass).

brianjmiller commented:
This code has landed as #56.
